scene/gui/control.cpp

```diff
@@ -2204,60 +2204,57 @@ bool Control::is_clipping_contents() {
 
 void Control::_propagate_theme_changed(Node *p_at, Control *p_owner, Window *p_owner_window, bool p_assign) {
 	Control *c = Object::cast_to<Control>(p_at);
-
-	if (c && c != p_owner && c->data.theme.is_valid()) { // has a theme, this can't be propagated
-		return;
+	if (c && c != p_owner && c->data.theme.is_valid()) {
+		// Has a theme, so we don't want to change the theme owner,
+		// but we still want to propagate in case this child has theme items
+		// it inherits from the theme this node uses.
+		// See https://github.com/godotengine/godot/issues/62844.
+		p_assign = false;
 	}
 
 	Window *w = c == nullptr ? Object::cast_to<Window>(p_at) : nullptr;
+	if (w && w != p_owner_window && w->theme.is_valid()) {
+		// Same as above.
+		p_assign = false;
+	}
 
-	if (w && w != p_owner_window && w->theme.is_valid()) { // has a theme, this can't be propagated
+	if (!c && !w) {
 		return;
 	}
 
 	for (int i = 0; i < p_at->get_child_count(); i++) {
-		CanvasItem *child = Object::cast_to<CanvasItem>(p_at->get_child(i));
-		if (child) {
-			_propagate_theme_changed(child, p_owner, p_owner_window, p_assign);
-		} else {
-			Window *window = Object::cast_to<Window>(p_at->get_child(i));
-			if (window) {
-				_propagate_theme_changed(window, p_owner, p_owner_window, p_assign);
-			}
-		}
+		_propagate_theme_changed(p_at->get_child(i), p_owner, p_owner_window);
 	}
 
 	if (c) {
 		if (p_assign) {
 			c->data.theme_owner = p_owner;
 			c->data.theme_owner_window = p_owner_window;
 		}
-		c->notification(Control::NOTIFICATION_THEME_CHANGED);
-		c->emit_signal(SceneStringNames::get_singleton()->theme_changed);
-	}
 
-	if (w) {
+		c->_notify_theme_changed();
+	} else if (w) {
 		if (p_assign) {
 			w->theme_owner = p_owner;
 			w->theme_owner_window = p_owner_window;
 		}
-		w->notification(Window::NOTIFICATION_THEME_CHANGED);
-		w->emit_signal(SceneStringNames::get_singleton()->theme_changed);
+
+		w->_notify_theme_changed();
 	}
 }
 
 void Control::_theme_changed() {
 	_propagate_theme_changed(this, this, nullptr, false);
 }
 
-void Control::_theme_property_override_changed() {
-	notification(NOTIFICATION_THEME_CHANGED);
-	emit_signal(SceneStringNames::get_singleton()->theme_changed);
-	update_minimum_size(); // Overrides are likely to affect minimum size.
```

Comment:
The `update_minimum_size()` is redundant as it's already called here:
https://github.com/godotengine/godot/blob/ba2aa30a1861aed55d1fc80fe48bc356afdf17bd/scene/gui/control.cpp#L3147